Query/EncloseNames.php

```diff
@@ -0,0 +1,132 @@
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Hoa nor the names of its contributors may be
+ *       used to endorse or promote products derived from this software without
+ *       specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS AND CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Hoa\Database\Query;
+
+/**
+ * Abstract Class \Hoa\Database\Query\EncloseNames.
+ *
+ * Enclose names feature.
+ *
+ * @copyright  Copyright © 2007-2015 Hoa community
+ * @license    New BSD License
+ */
+abstract class EncloseNames
+{
+    /**
+     * State of the enclose names feature.
+     *
+     * @var bool
+     */
+    protected $enableEnclose = false;
+
+    /**
+     * Enclose open symbol.
+     *
+     * @var string
+     */
+    protected $openSymbol    = '"';
+
+    /**
+     * Enclose close symbol.
+     *
+     * @var string
+     */
+    protected $closeSymbol   = '"';
+
+
+
+    /**
+     * Set enclose symbols.
+     *
+     * @param  string  $openSymbol     Open symbol.
+     * @param  string  $closeSymbol    Close symbol.
+     * @return  static
+     */
+    public function setEncloseSymbol($openSymbol, $closeSymbol = null)
+    {
+        $this->openSymbol  = $openSymbol;
+        $this->closeSymbol = (null === $closeSymbol)
+            ? $openSymbol
+            : $closeSymbol;
+
+        return $this;
+    }
+
+    /**
+     * Enable enclose names feature.
+     *
+     * @return  static
+     */
+    public function enableEncloseNames()
+    {
+        $this->enableEnclose = true;
+
+        return $this;
+    }
+
+    /**
+     * Disable enclose names feature.
+     *
+     * @return  static
+     */
+    public function disableEncloseNames()
+    {
+        $this->enableEnclose = false;
+
+        return $this;
+    }
+
+    /**
+     * Enclose a table/column/alias name with defined symbol.
+     *
+     * @param  array|string  $name    The table/column/alias name.
+     * @return  string
+     */
+    protected function enclose($name)
+    {
+        if (false === $this->enableEnclose) {
+            return $name;
+        }
+
+        if (is_array($name)) {
+            return array_map([$this, __FUNCTION__], $name);
```

Comment:
In this way you will always return an array which is not expected.
So, IMO It's preferable to duplicate the enclosing code inside the if.

```php
if (false === is_array($names)) {
    return $this->_openingSymbol . $names . $this->_closingSymbol;
}

foreach …
```